### ISSUE TITLE
Following start guide - how to view created resources in AWS console?

### ISSUE THREAD

**Issue by andrew89898** [awslabs/fleetiq-adapter-for-agones#8]
Apologies for the newbie question - I am learning about Agones, GameLift FleetIQ and plan to use AWS FlexMatch to create a game service for CSGO (source engine) - I plan to use a webhook autoscaler with AWS and write a wrapper for the game server.

I followed the guide to create a workspace and set up all the aws resources - all good managed to get to the end of the guide and proceed with the scaling up and replicas seeing the nodes - however it didn't ever scale up to 100 nodes stopping at 54 with 1 group and 60 with 2 groups. I'd like to understand more of this adapter and agones but I can't see any of these resources in my AWS account except for the GameLift groups. The EKS cluster does show up but doesn't display any nodes.

This isn't really an issue with the repo - I'd just like to learn more but I can't see any forum or slack channel/similar for this. I'd like to be able to work on the components of the solution architecture and visualise where I need to work on, if you guys can give any recommendations.

Thanks.

**Comment by jicowan** [comment]
@andrew89898 you may not have permission to view the nodes in your EKS cluster from the EKS console. Can you run `kubectl get nodes` and add the output from that command to this issue? If you've installed the Kubernetes cluster autoscaler, please include the logs from that pod too, e.g. `kubectl logs <podname> -n <namespace>`. I'd also like to know how many instances in the ASG (if any) are in a standby state.

**Comment by andrew89898** [comment]
Sure can do:

`ec2-user:~/environment $ kubectl get nodes
NAME                                           STATUS   ROLES    AGE     VERSION
ip-192-168-2-176.eu-west-2.compute.internal    Ready    <none>   3d22h   v1.18.9-eks-d1db3c
ip-192-168-23-13.eu-west-2.compute.internal    Ready    <none>   3d20h   v1.18.9-eks-d1db3c
ip-192-168-41-125.eu-west-2.compute.internal   Ready    <none>   6h20m   v1.18.9-eks-d1db3c
ip-192-168-60-223.eu-west-2.compute.internal   Ready    <none>   3d22h   v1.18.9-eks-d1db3c
ec2-user:~/environment $ `

Logs (lots of lines, got first page):
(command is: `kubectl logs cluster-autoscaler-57bdc587bf-v66tp -n kube-system`
`E0511 17:45:00.878519       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:45:04.979515       1 reflector.go:496] k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:246: Watch close - *v1.Node total 7 items received
I0511 17:45:10.878649       1 static_autoscaler.go:226] Starting main loop
E0511 17:45:10.878891       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:45:10.878903       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:45:20.879035       1 static_autoscaler.go:226] Starting main loop
E0511 17:45:20.879272       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:45:20.879284       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:45:26.996389       1 node_instances_cache.go:156] Start refreshing cloud provider node instances cache
E0511 17:45:26.996420       1 node_instances_cache.go:164] Failed to get cloud provider node instance for node group gamelift-gameservergroup-, error error while looking for instances of ASG: {gamelift-gameservergroup-}
I0511 17:45:26.996428       1 node_instances_cache.go:168] Refresh cloud provider node instances cache finished, refresh took 18.977µs
I0511 17:45:30.879414       1 static_autoscaler.go:226] Starting main loop
E0511 17:45:30.879661       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:45:30.879674       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:45:37.179538       1 reflector.go:496] k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:309: Watch close - *v1beta1.PodDisruptionBudget total 0 items received
I0511 17:45:40.879802       1 static_autoscaler.go:226] Starting main loop
E0511 17:45:40.880032       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:45:40.880043       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:45:50.880186       1 static_autoscaler.go:226] Starting main loop
E0511 17:45:50.880445       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:45:50.880458       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:46:00.880606       1 static_autoscaler.go:226] Starting main loop
I0511 17:46:00.880659       1 auto_scaling_groups.go:351] Regenerating instance to ASG map for ASGs: [gamelift-gameservergroup- gamelift-gameservergroup-agones-game-server-group-02]
I0511 17:46:00.942462       1 auto_scaling.go:199] 1 launch configurations already in cache
I0511 17:46:00.942486       1 auto_scaling_groups.go:114] Updating ASG gamelift-gameservergroup-agones-game-server-group-02
I0511 17:46:00.942495       1 aws_manager.go:269] Refreshed ASG list, next refresh after 2021-05-11 17:47:00.942492397 +0000 UTC m=+17643.158739587
E0511 17:46:00.942694       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:46:00.942705       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:46:10.942867       1 static_autoscaler.go:226] Starting main loop
E0511 17:46:10.943115       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:46:10.943126       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:46:13.683475       1 reflector.go:496] k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:212: Watch close - *v1.Pod total 4 items received
I0511 17:46:20.943283       1 static_autoscaler.go:226] Starting main loop
E0511 17:46:20.943537       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:46:20.943548       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:46:30.943707       1 static_autoscaler.go:226] Starting main loop
E0511 17:46:30.943952       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:46:30.943964       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:46:40.944109       1 static_autoscaler.go:226] Starting main loop
E0511 17:46:40.944343       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:46:40.944354       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:46:44.389064       1 reflector.go:496] k8s.io/autoscaler/cluster-autoscaler/utils/kubernetes/listers.go:320: Watch close - *v1.DaemonSet total 3 items received
I0511 17:46:50.944493       1 static_autoscaler.go:226] Starting main loop
E0511 17:46:50.944732       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:46:50.944744       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:47:00.944886       1 static_autoscaler.go:226] Starting main loop
I0511 17:47:00.944932       1 auto_scaling_groups.go:351] Regenerating instance to ASG map for ASGs: [gamelift-gameservergroup-agones-game-server-group-02 gamelift-gameservergroup-]
I0511 17:47:01.039568       1 auto_scaling.go:199] 1 launch configurations already in cache
I0511 17:47:01.039592       1 auto_scaling_groups.go:114] Updating ASG gamelift-gameservergroup-agones-game-server-group-02
I0511 17:47:01.039600       1 aws_manager.go:269] Refreshed ASG list, next refresh after 2021-05-11 17:48:01.039597629 +0000 UTC m=+17703.255844816
E0511 17:47:01.039844       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:47:01.039856       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:47:04.380433       1 reflector.go:496] k8s.io/client-go/informers/factory.go:135: Watch close - *v1.PersistentVolume total 0 items received
I0511 17:47:11.040029       1 static_autoscaler.go:226] Starting main loop
E0511 17:47:11.040299       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:47:11.040312       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:47:21.040466       1 static_autoscaler.go:226] Starting main loop
E0511 17:47:21.040714       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:47:21.040726       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:47:26.996557       1 node_instances_cache.go:156] Start refreshing cloud provider node instances cache
E0511 17:47:26.996589       1 node_instances_cache.go:164] Failed to get cloud provider node instance for node group gamelift-gameservergroup-, error error while looking for instances of ASG: {gamelift-gameservergroup-}
I0511 17:47:26.996598       1 node_instances_cache.go:168] Refresh cloud provider node instances cache finished, refresh took 22.646µs
I0511 17:47:31.040886       1 static_autoscaler.go:226] Starting main loop
E0511 17:47:31.041132       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:47:31.041144       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:47:41.041290       1 static_autoscaler.go:226] Starting main loop
E0511 17:47:41.041529       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:47:41.041541       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:47:51.041679       1 static_autoscaler.go:226] Starting main loop
E0511 17:47:51.041925       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:47:51.041936       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
I0511 17:48:01.042324       1 static_autoscaler.go:226] Starting main loop
I0511 17:48:01.042393       1 auto_scaling_groups.go:351] Regenerating instance to ASG map for ASGs: [gamelift-gameservergroup-agones-game-server-group-02 gamelift-gameservergroup-]
I0511 17:48:01.199847       1 auto_scaling.go:199] 1 launch configurations already in cache
I0511 17:48:01.199873       1 auto_scaling_groups.go:114] Updating ASG gamelift-gameservergroup-agones-game-server-group-02
I0511 17:48:01.199883       1 aws_manager.go:269] Refreshed ASG list, next refresh after 2021-05-11 17:49:01.199880066 +0000 UTC m=+17763.416127279
E0511 17:48:01.200090       1 utils.go:119] Unable to build proper template node for gamelift-gameservergroup-: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
E0511 17:48:01.200103       1 static_autoscaler.go:268] Failed to get node infos for groups: unable to get first AvailabilityZone for ASG "gamelift-gameservergroup-"
ec2-user:~/environment $ `

**Comment by jicowan** [comment]
So it looks like you have 2 ASGs/game server groups:

 --nodes=0:10:gamelift-gameservergroup-
 --nodes=0:10:gamelift-gameservergroup-agones-game-server-group-02

It also looks like you're using the priority expander. I suggest deleting 1 of the 2 ASGs and using the least waste expander with auto discovery. Not sure why the guide has you create multiple game server groups. You should be able to get the capacity you need from a single ASG with diverse instance types that spans multiple AZs.

**Comment by andrew89898** [comment]
Thanks @jicowan I will look into deleting the groups. At this stage I am just testing agones and need to develop a wrapper and autoscaler - Ideally I'd like to run this for cheap or locally to avoid charges - following the guide with no mention of costs, I have accrued $200+ in charges! I just wanted to learn about the system.

Any advice?

**Comment by jicowan** [comment]
How long has the environment been running? The EKS control plane costs 10 cents an hour which comes to about $70/mo. You also pay for the provisioned compute. A majority of that should be Spot which costs 40% less (on average) than the on demand cost. FleetIQ doesn't offer as steep a discount as Spot. If you're primarily interested in testing Agones, you can try using Kubernetes in Docker (kind) assuming you have a machine powerful enough to run your game + Kubernetes. Definitely appreciate you trying the solution. Happy to continue working with you to get it working in your environment.

**Comment by jicowan** [comment]
You need to delete the game server group to delete the ASGs. You can't delete the ASGs directly because they are managed by FleetIQ.

**Comment by andrew89898** [comment]
Think its been running since May 7th, I'm in the GameLift game server group area on the console and have no options to delete, might be permissions. Are the some commands I could run in the Cloud9 dev environment?

**Comment by jicowan** [comment]
aws gamelift delete-game-server-group --game-server-group-name <your gsg name>

I think that'll do it.

**Comment by andrew89898** [comment]
Yeah that should do it, but I don't know how to force or shutdown the gameservers first?
```
ec2-user:~/environment $ aws gamelift delete-game-server-group --game-server-group-name agones-game-server-group-01

An error occurred (InvalidRequestException) when calling the DeleteGameServerGroup operation: Failed to SAFE_DELETE as there exists at least one UTILIZED game server
ec2-user:~/environment $ aws gamelift delete-game-server-group --game-server-group-name agones-game-server-group-02

An error occurred (InvalidRequestException) when calling the DeleteGameServerGroup operation: Failed to SAFE_DELETE as there exists at least one UTILIZED game server
ec2-user:~/environment $ 
```

**Comment by jicowan** [comment]
No problem. Go to the ASG and set the min and desired count to 0 first.

**Comment by andrew89898** [comment]
Thanks - managed to set the ASG programatically with aws autoscale, but then I found the groups in the console anyway and have terminated the instances. Also found some ELBs. My services say I am still billed for EKS and some GameLift things but in those respective areas of the console I can't view anything to delete, even as the root user.

I've got a support case going on so they might be able to help me further.
